sr_multi_moveit/sr_multi_moveit_config/scripts/generate_moveit_config.py

```diff
@@ -0,0 +1,408 @@
+            output_str += "    joints:\n"
+            if len(group.joints) == 0:
+                output_str += "      []\n"
+            else:
+                for joint in group.joints:
+                    if joint.name[-3:] != "tip":
+                        if manipulator.has_arm:
+                            if joint.name[len(manipulator.hand.prefix):] not in ["WRJ1", "WRJ2"]:
+                                output_str += "      - " + joint.name + "\n"
+                        else:
+                            output_str += "      - " + joint.name + "\n"
+
+    # load on param server or output to file
+    upload_output_params(output_str, output_path, ns_)
+
+
+def generate_real_controllers(robot, robot_config, output_path=None, ns_=None):
+    output_str = ""
+    output_str += "controller_list:\n"
+    for manipulator in robot_config.manipulators:
+        if manipulator.has_arm:
+            # Read arm srdf
+            arm_yaml_path = manipulator.arm.moveit_path + "/" + "controllers.yaml"
+            with open(arm_yaml_path, 'r') as stream:
+                arm_yamldoc = yaml.load(stream)
+
+            output_str += "  - name: " + manipulator.arm.prefix + arm_yamldoc["controller_list"][0]["name"] + "\n"
+            output_str += "    action_ns: follow_joint_trajectory\n"
+            output_str += "    type: FollowJointTrajectory\n"
+            output_str += "    default: true\n"
+            output_str += "    joints:\n"
+            for joint in arm_yamldoc["controller_list"][0]["joints"]:
+                output_str += "     - " + manipulator.arm.prefix + joint + "\n"
+            if manipulator.has_hand and not manipulator.hand.is_lite:
+                output_str += "     - " + manipulator.hand.prefix + "WRJ2" + "\n"
+                output_str += "     - " + manipulator.hand.prefix + "WRJ1" + "\n"
+
+        if manipulator.has_hand:
+            sh_group = None
+            for group in robot.groups:
+                if group.name == manipulator.hand.internal_name:
+                    sh_group = group
+                    break
+            controller_name = "  - name: " + manipulator.hand.prefix + "trajectory_controller\n"
+            output_str += controller_name
+            output_str += "    action_ns: follow_joint_trajectory\n"
+            output_str += "    type: FollowJointTrajectory\n"
+            output_str += "    default: true\n"
+            output_str += "    joints:\n"
+            if len(group.joints) == 0:
+                output_str += "      []\n"
+            else:
+                for joint in group.joints:
+                    if joint.name[-3:] != "tip":
+                        if manipulator.has_arm:
+                            if joint.name[len(manipulator.hand.prefix):] not in ["WRJ1", "WRJ2"]:
+                                output_str += "      - " + joint.name + "\n"
+                        else:
+                            output_str += "      - " + joint.name + "\n"
```

Comment:
how do these 2 statements differ, looks like if and else result in the same outcome?